src/HIE/Bios/Cradle.hs

```diff
@@ -787,34 +794,39 @@ cabalAction
   -> LogAction IO (WithSeverity Log)
   -> CradleProjectConfig
   -> FilePath
-  -> [FilePath]
+  -> LoadStyle
   -> CradleLoadResultT IO ComponentOptions
-cabalAction (ResolvedCradles root cs vs) workDir mc l projectFile fp fps = do
+cabalAction (ResolvedCradles root cs vs) workDir mc l projectFile fp loadStyle = do
   cabal_version <- liftIO $ runCachedIO $ cabalVersion vs
   ghc_version   <- liftIO $ runCachedIO $ ghcVersion vs
   let
     cabalCommand = "v2-repl"
-    cabalArgs = case (cabal_version, ghc_version) of
-      (Just cabal, Just ghc)
-        -- Multi-component supported from cabal-install 3.11
-        -- and ghc 9.4
-        | ghc   >= makeVersion [9,4]
-        , cabal >= makeVersion [3,11]
-        -> case fps of
-          [] -> [fromMaybe (fixTargetPath fp) mc]
-          -- Start a multi-component session with all the old files
-          _ -> "--keep-temp-files"
-             : "--enable-multi-repl"
-             : fromMaybe (fixTargetPath fp) mc
-             : [fromMaybe (fixTargetPath old_fp) old_mc
-               | old_fp <- fps
-               -- Lookup the component for the old file
-               , Just (ResolvedCradle{concreteCradle = ConcreteCabal ct}) <- [selectCradle prefix old_fp cs]
-               -- Only include this file if the old component is in the same project
-               , (projectConfigFromMaybe root (cabalProjectFile ct)) == projectFile
-               , let old_mc = cabalComponent ct
-               ]
-      _ -> [fromMaybe (fixTargetPath fp) mc]
+  cabalArgs <- case (cabal_version, ghc_version) of
+    (Just cabal, Just ghc)
+      -- Multi-component supported from cabal-install 3.11
+      -- and ghc 9.4
+      | ghc   >= makeVersion [9,4]
+      , cabal >= makeVersion [3,11]
+      -> case loadStyle of
+        SingleComponent -> pure [fromMaybe (fixTargetPath fp) mc]
+        -- Start a multi-component session with all the old files
+        LoadWithContext fps -> do
+          liftIO $ l <& LogCabalMultiComponentSupport `WithSeverity` Info
```

Comment:
Okay, I see we're doing something here.

I guess maybe what I'd be inclined to do would be something like:
```
log $ RequestedLoadStyle userLoadStyle
determinedLoadStyle <- case (Just cabal, Just ghc) of ...
log $ DeterminedLoadStyle determinedLoadStyle
case determinedLoadStyle of ...
```

Reply:
Also I don't think you're going to get anything useful if you pick the multi-component load style with a stack cradle.

Reply:
Yeah, but we want all cradles to have the same interface. Otherwise, it gets tricky real quickly and this model has served us well, so far.

Reply:
Right, I just want to get some kind of clear log output indicating that the option is doing nothing in that case.

Reply:
Indeed, that's a good idea! Changed the logging to log in the case of `stack` and `bios` cradle that the load style is not supported.